pkg/ibmvpc/parser.go

```diff
@@ -380,27 +381,58 @@ func (rc *IBMresourcesContainer) getInstancesConfig(
 		vpcConfig := res.Config(vpcUID)
 		vpcConfig.NodeSets = append(vpcConfig.NodeSets, vsiNode)
 		vpcConfig.UIDToResource[vsiNode.ResourceUID] = vsiNode
+		// a VSI can not have both VNIs and NIs.
+		// first we check for VNIs.
+		// the VNIs are listed at the instance.NetworkAttachments.
+		// using that list, we extract the VNIs from the rc.VirtualNIList, which holds all the VNIs of all VSIs
+		for j := range instance.NetworkAttachments {
+			vniObj := vnisObj[*instance.NetworkAttachments[j].ID]
```

Comment:
Need to check we do not get a `nil` here (might happen when running with input files made by an older version of the collector)

Reply:
we can range over nil on golang

Reply:
I meant what if `vnisObj[*instance.NetworkAttachments[j].ID]` returns `nil`...

Reply:
in such case, we have a vins that we can not parse.
we can not use the nis instead, since their ids are not at the filters rules
fix to return error